### ISSUE TITLE
Add the `mandatory_domain` field in `src/endpoints/admin/quest/update_quest.rs`

### ISSUE THREAD

**Issue by Marchand-Nicolas** [starknet-id/quest_server#364]
## Description 📹

**Add the `mandatory_domain` field in `src/endpoints/admin/quest/update_quest.rs`, which should be either `null` or a `string`. Update the necessary structures and logic to include this new field.**

## Proposed Actions 🛠️

Here’s a checklist of actions to follow for resolving this issue:

1. **Fork and Create Branch**:  
   Fork the repository and create a new branch using the issue number:

```bash
   git checkout -b fix-[issue-number]
```

2. **Implement Changes**:  
   Update the code to include the `mandatory_domain` field.

   - [ ] Add the `mandatory_domain` field in the `UpdateQuestQuery` struct:
     
     ```rust
     pub_struct!(Deserialize; UpdateQuestQuery {
         id: i32,
         name: Option<String>,
         desc: Option<String>,
         start_time: Option<i64>,
         expiry: Option<i64>,
         disabled: Option<bool>,
         category: Option<String>,
         logo: Option<String>,
         rewards_img: Option<String>,
         rewards_title: Option<String>,
         img_card: Option<String>,
         title_card: Option<String>,
         issuer: Option<String>,
         mandatory_domain: Option<String>, // New field
     });
     ```

   - [ ] Update the `update_doc` initialization to include `mandatory_domain`:
     
     ```rust
     if let Some(mandatory_domain) = &body.mandatory_domain {
         update_doc.insert("mandatory_domain", mandatory_domain);
     }
     ```

   - [ ] Ensure the `mandatory_domain` is appropriately updated if provided, or skipped if `null`.

3. **Run Tests and Commit Changes**:  
   Make sure your changes don't break existing functionality and commit with a clear message:

```bash
   git commit -m "Fix: Add mandatory_domain field to UpdateQuestQuery and update logic"
```

## Required 📋

To keep our workflow smooth, please make sure you follow these guidelines:

- **Assignment**: Don't create a pull request if you weren’t assigned to this issue.
- **Timeframe**: Complete the task within **3 business days**.
- **Closing the Issue**: In your PR description, close the issue by writing Close #[issue_id].
- **Review Process**:
  - Once you've submitted your PR, change the label to **"ready for review"**.
  - If changes are requested, address them and then update the label back to **"ready for review"** once done.
- **Testing**: Test your PR locally before pushing, and verify that tests and build are working after pushing.

Thank you for your contribution 🙏

⚠️ WARNING: Failure to follow the requirements above may result in being added to the OnlyDust blacklist, affecting your ability to receive future rewards.

**Comment by Benjtalkshow** [comment]
Can I handle this?

**Comment by No-bodyq** [comment]
May I take care of this?

**Comment by ryzen-xp** [comment]
Is it okay if I tackle this?

**Comment by Supa-mega** [comment]
Can I take this issue?

**Comment by VyuduInc** [comment]
Can I be assigned to this?

**Comment by mimisavage** [comment]
I'd like to help with this.

**Comment by 847850277** [comment]
Can I tackle this one?

**Comment by mahaitao617** [comment]
I’d love to work on this task.

**Comment by elmuby** [comment]
Is it okay if I tackle this?

**Comment by Nityam573** [comment]
I'd be happy to do this.

**Comment by cryptogru725** [comment]
I'd love to give this a go.

**Comment by aurlic** [comment]
I’d like to work on this.

**Comment by Thibrac** [comment]
Can I attempt this issue?

**Comment by emarc99** [comment]
Mind if I take this issue?

**Comment by thedevtask** [comment]
Mind if I try this one?

**Comment by pcarette** [comment]
I worked on it and i'm ready to PR, can you assign me the task ?

**Comment by olisaagbafor** [comment]
Can I tackle this one?

**Comment by crood-xp** [comment]
Can I start working on this?

**Comment by jaykayudo** [comment]
Can i tackle this ? @Marchand-Nicolas 

## My Background
I am a full stack developer with experience and I specialize in building dynamic and scalable software applications using a combination of Python, Rust , Javascript, and Node. 

### My Approach to tackling this issue
- Fork and create a branch based on the issue number
- Add the `mandatory_domain` field in `src/endpoints/admin/quest/update_quest.rs` file `UpdateQuestQuery` struct.
- Add a check for the `mandatory_domain` and add it to the update_doc if it is not None.
- Run tests to ensure that there is no breaking change.
-  Commit my change using the `git commit -m "Fix: Add mandatory_domain field to UpdateQuestQuery and update logic"` syntax
- In summary, I will follow the proposed instruction in this issue description to the letter.
- Open a PR within 24 hours

**Comment by DaPrice1** [comment]
I'd love to work on Hello everyone, greetings and salutations, I'm a New contributor and I would really love to contribute to this project, only because I know and I am truly more than capable enough.

DaPrice1

**Comment by No-bodyq** [comment]
Let me try this one!

**Comment by akintewe** [comment]
Hello good day, i have experience in Rust, I've worked with Starknet APIs before and would love to help add the mandatory_domain field to the quest update endpoint. I can ensure proper integration while preserving the existing codebase patterns. My ETA is 8 hours
